lib/components/buttons/buttonContainer.dart

```diff
@@ -19,19 +22,43 @@ class ButtonContainer extends StatefulWidget {
 
 class _ButtonContainer extends State<ButtonContainer> {
   final Set<String> achivements = Set.from(['1.1']);
-  bool clicksCompleted = false;
+  bool isClicksCompleted = false;
+  bool isSwipesUpCompleted = false;
+  bool isSwipesDownCompleted = false;
+  bool isSwipesLeftCompleted = false;
+  bool isSwipesRightCompleted = false;
 
+  @override
+  void initState() {
+    super.initState();
+    String challangeKey = AppStore.state.challengeKey;
+    print(challangeKey);
+    setCompletedChecks(challangeKey);
+  }
+
+  void setCompletedChecks(String challangeKey) {
+    Map<String, bool> availableActions = getAvailableActions(challangeKey);
+    isClicksCompleted = !availableActions['clicks'];
+    isSwipesUpCompleted = !availableActions['swipesUp'];
+    isSwipesDownCompleted = !availableActions['swipesDown'];
+    isSwipesLeftCompleted = !availableActions['swipesLeft'];
+    isSwipesRightCompleted = !availableActions['swipesRight'];
+  }
   /*
    * runs on each interaction with the button
    * should reset store and change challange if all
    * conditions are true
   */
-  void onChallangeCompletion (_ViewModel vm) {
-    if (this.clicksCompleted) {
-      vm.resetClickCount();
-      vm.setChallengeKey(getNextChallengeKey(vm.challengeKey));
+  void checkChallangeCompletion(_ViewModel vm) {
```

Comment:
challenge*

Reply:
:(

Reply:
We fix it later it's misspelled on other earlier places as-well xD